### ISSUE TITLE
Responsive - drags can't change between breakpoints

### ISSUE THREAD

**Issue by tbb2** [OwlCarousel2/OwlCarousel2#1559]
mouseDrag, touchDrag and pullDrag are not working in responsive mode. In the below example, if you load the website in the largest breakpoint, drag is not enabled when you go to the lower breakpoint. 

$(".owl-carousel").owlCarousel({
    responsive:{
        0:{
            items:1,
            nav:true,
            mouseDrag:true,
            touchDrag:true,
            pullDrag:true
        },
        800:{
            items:3,
            nav:false,
            mouseDrag:false,
            touchDrag:false,
            pullDrag:false
        }
    }
});

**Comment by daviddeutsch** [comment]
(Sorry about the canned reply.)

Please add a link to a page where I can verify this error. You can use a service like [jsfiddle](http://jsfiddle.net/), [plnkr](https://plnkr.co/), [jsbin](http://jsbin.com/) or [codepen](http://codepen.io/).

And make sure you're using the latest Owl Version!

**Comment by jacobkossman** [comment]
http://codepen.io/jkingaround/pen/RpoxEL

This is a codepen demonstrating the same issue occurring on the latest release (2.2.1) at the time of writing this. 

Reproduce by resizing the main page until it shows 2 items and you can't drag. A refresh in the same size window enables drag but it does not change upon resizing the window unlike number of items which works fine.

Thanks!

**Comment by mtaltavull** [comment]
https://codepen.io/mtaltavull/pen/VxyVVM

I am also able to reproduce this on 2.3.4.

1. Visit the codepen with a browser width greater than 1200. Mouse drag will be disabled as expected.
2. Reduce the window size < 1200. Mouse drag will still be disabled even though the responsive settings are setting mouseDrag equal to true. 
3. Reloading the browser with a browser width < 1200 will cause the mouseDrag setting to be respected.

**Comment by pascalporedda** [comment]
I can approve that.

**Comment by marcelo2605** [comment]
Having same problem here. Any solution?

**Comment by VRa3** [comment]
Would be cool if anyone could provide a solution for this.

I think there is listener needed for window resizing.

**Comment by princefishthrower** [comment]
@VRa3 @marcelo2605 @pascalporedda @mtaltavull 

Old, old, old issue - but this should be fixed in this PR:

https://github.com/OwlCarousel2/OwlCarousel2/pull/2666/files#diff-09e8e94a19a18a4c4155a5385b34ef15R594

tl;dr; add the following to the `updateLogic` function:

```javascript
if (this.settings.mouseDrag) {
    this.$element.addClass(this.options.dragClass);
    this.$stage.on('mousedown.owl.core', $.proxy(this.onDragStart, this));
    this.$stage.on('dragstart.owl.core selectstart.owl.core', function() { return false });
} else {
    this.$element.removeClass(this.options.dragClass);
    this.$element.removeClass(this.options.grabClass);
    this.$stage.off('mousedown.owl.core');
    this.$stage.off('dragstart.owl.core selectstart.owl.core');
}
if (this.settings.touchDrag){
    this.$stage.on('touchstart.owl.core', $.proxy(this.onDragStart, this));
    this.$stage.on('touchcancel.owl.core', $.proxy(this.onDragEnd, this));
} else {
    this.$stage.off('mousedown.owl.core');
    this.$stage.off('touchcancel.owl.core');
}
```

**Comment by yurychang** [comment]
I think this line on bottom was wrong:
`this.$stage.off('touchcancel.owl.core');`
change to:
`this.$stage.off('touchstart.owl.core');`
